fix: Allow DELETE queries to run on Presto on Spark

ghelmling opened the pull request:
## Description
Simple change to allow DELETE queries to run in Presto-on-Spark for connectors that implement them

Reviewed By: shelton408

Differential Revision: D83402171

## Release Notes
```
== RELEASE NOTES ==
General Changes
* Fix a gap in query commit for DELETE queries when running on Spark.
```

tdcmeehan commented:
Can we write a test that shows it works?

singcha commented:
> Can we write a test that shows it works?

@GaryHo34  - You can add a test with delete query here - TestPrestoSparkQueryRunner.java

ghelmling committed:
Allow DELETE queries to run on Presto on Spark

Summary: Simple change to allow DELETE queries to run in Presto-on-Spark for connectors that implement them

Reviewed By: shelton408

Differential Revision: D83402171

steveburnett commented:
* Please edit the PR title to follow semantic commit style to pass the failing and required CI check. See the failure in the test for advice. 

* Please add a release note - or `NO RELEASE NOTE` - following the [Release Notes Guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines) to pass the failing but not required CI check.
